### ISSUE TITLE
Make use of CPU samples

### ISSUE THREAD

**Issue by dominikh** [dominikh/gotraceui#16]
Go 1.19 adds CPU samples to runtime traces. This gives us multiple stacks per span. We should expose those somehow. The most straightforward would be to expose them like other events. It's also tempting to recreate timelines from samples, but we have to be clear about the sampled nature; a single sample doesn't tell us if a function got called once or a dozen times, and if the call lasted a nanosecond or 10ms.

**Comment by dominikh** [comment]
For now I'm going with the "recreating timelines" approach. Plenty of other tracers do this, too, so our users should be able to handle its inaccuracies.

We could also enrich our sample-based timelines with stack traces from runtime events. During a single 10ms span (i.e. one CPU sample), we may have access to many more stack traces of events that occurred during the sample, such as preemption, blocking/unblocking etc. This should probably be an opt-in feature, though. Turning a single CPU sample into a 10ms span is already making it tricky to interpret the data correctly. If we mix this with more accurate events then users are even more prone to assuming that the entire sampled timeline is accurate, or that a 9.8ms long span (the remainder of a sample) _definitely_ was longer than a 0.2ms span (a span resulting from incorporating traced events).

Screenshot of the current WIP:
![image](https://user-images.githubusercontent.com/39825/191137161-3cefb2f6-d8fa-487b-8ec3-9c37264fc47c.png)

**Comment by dominikh** [comment]
At the same time it looks like we might _have to_ incorporate traced events. I'm still investigating, but at first glance it looks like we're not getting samples for goroutines that are blocked. This leads to the following screenshot:

![image](https://user-images.githubusercontent.com/39825/191138363-51a3e439-ceb9-4d2b-ad70-9f007922b414.png)

I've disabled span merging so that each individual sample is visible. During the grey regions we're sleeping, and we don't seem to be receiving any samples, making it look like we're spending a lot of time in crypto/sha1.Sum, while in reality we've returned from it and are sleeping a lot.
